readme.md

```diff
@@ -1,9 +1,224 @@
+
+
+## Filtering routine overview
+
+Here goes an overview of the general filtering routine. For more details, check the code [documentation](docs/src/main/scala/). Filtering is split in several steps:
+
+- `filter1`: filters out unrelated and uninformative *sequences*
+- `filter2`: filters out uninformative *assignments*
+- `filter3`: filters out *wrong* assignments
+
+Some more details about each step.
+
+### `filter1`
+
+This is an important filter, because out of the huge RNAcentral database it chooses only 16S sequences and filters out those that are badly classified or are not informative.
+To pass this filter sequences have to satisfy the following predicate:
+
+- the sequence ID is **not** in `blacklistedRNACentralIDs`
+- the corresponding sequence is longer than `minimum16SLength`
+- for each taxonomy association separately:
+  + annotated as an `rRNA`
+  + **not** one of those in `uninformativeTaxIDs`
+  + a descendant of either *Archaea* or *Bacteria*
+  + **not** a descendant of an *environmental* or *unclassified* taxon
+
+You can lookup the mentioned constants in the [`filter1`](docs/src/main/scala/filter1.scala.md) code.
+
+Only **~3.9%** of all RNAcentral sequences pass this filter.
+
+
+### `filter2`
+
+This step filters out assignments that are covered by bigger sequences. For example, if a sequence `S` has an assignment to taxon `A`, and a sequence `s` which is a subsequence of `S` has the same assignment, it gets discarded for `s`. If this was the only assignment for `s`, then the sequence gets discarded.
+
+A match with a bigger sequence is always better, but if we won't discard those subsequence assignments they will be always implied, which doesn't add information for the analysis. So this filter allows to reduce the size of the database while leaving only most informative assignments.
+
+After this step a BLAST database is generated from the sequences that passed the filter.
+
+Around **72%** of the sequences from `filter1` pass `filter3`. Among these sequences there are also some with a reduced number of assignments.
+
+
+### `filter3`
+
+This step actually consists of two:
+
+- Run MG7 pipeline using the reference database from `filter2` using its output sequences FASTA as input data. This will produce a table which matches each sequence the lowest common ancestor (LCA) of the assignments of any other (very) similar sequences.
+
+- For each sequence we check the relation of its assignments with the corresponding LCA that we've got from MG7. If some assignment is too far away from the LCA in the taxonomic tree, it is discarded.  
+  If a sequence has a single assignment, it is ignored, because it may represent a rare organism and it may be misplaced in the taxonomic tree.
+
```

Comment:
It tries to say too many things at the same time 😅 
If a sequence has only one (single) assignment, it is not tested with the described filter, because
- it may represent some rare organism that doesn't necessarily have relations with its taxonomic neighbours
- it can be an organism that was originally misclassified and put in the taxonomy far away from its real relatives (that could be discovered later, for example). This is how I understood what @rtobes told me.